packages/docusaurus-plugin-content-pages/src/index.ts

```diff
@@ -66,15 +71,22 @@ export default function pluginContentPages(
         return;
       }
 
-      const {addRoute} = actions;
+      const {addRoute, createData} = actions;
 
+      const configPath = await createData(
+        `${docuHash('config')}.json`,
+        JSON.stringify({siteConfig: context.siteConfig}, null, 2),
+      );
```

Comment:
Unfortunately, this works but is not ideal. We should try to reuse the existing config file of the .docusaurus folder.

If you look at App.tsx it imports it directly: `import siteConfig from '@generated/docusaurus.config';`

We should avoid generating a new json file containing exactly the same data.

I don't know exactly how to do this for now, this may be more complex than I thought.

Some idea I have is to add a "@theme/internal/PageHandler" component, that would call `useDocusaurusConfig`  and inject it into the page.

Maybe there's a simpler option, but I don't know which one for now. If you want to explore an alternative solution that does not call createData() that would be nice ;)

Reply:
I had the same idea but then thought if `docusaurus.config` file is a core dependency we shouldn't use it.